version: "3.8"

services:
  client:
    build: .
    image: benefits_client:latest
    environment:
      - DJANGO_SECRET_KEY
      - DJANGO_DEBUG
      - DJANGO_ADMIN
      - DJANGO_DB
      - DJANGO_INIT_JSON
      - DJANGO_INIT_PATH
      - GUNICORN_CMD_ARGS
    ports:
      - "${DJANGO_LOCAL_PORT}:80"

  dev:
    build:
      context: .
      dockerfile: Dockerfile.dev
    image: benefits_client:dev
    environment:
      - DJANGO_SECRET_KEY
      - DJANGO_DEBUG
      - DJANGO_ADMIN
      - DJANGO_DB
      - DJANGO_INIT_JSON
      - DJANGO_INIT_PATH=data/client/test.json
    entrypoint: []
    command: python -m debugpy --wait-for-client --listen 0.0.0.0:5678 manage.py runserver 0.0.0.0:8888
    depends_on:
      - server
    ports:
      - "${DJANGO_LOCAL_PORT}:8888"
      - "5678:5678"
    volumes:
      - .:/home/calitp/app:cached

  server:
    build: ./server
    image: eligibility_verification_server:dev
    command: python -m debugpy --listen 0.0.0.0:5789 app.py
    environment:
      - FLASK_APP=app.py
      - FLASK_DEBUG=1
      - FLASK_ENV=development
    ports:
      - "5000:5000"
      - "5789:5789"
    volumes:
      - ./server:/usr/src/server:cached
      - ./data/server:/usr/src/server/data
      - ./keys:/usr/src/server/keys
